build.sbt

```diff
@@ -348,41 +349,63 @@ def githubUrl(v: String): String = {
   "https://github.com/lightbend/kalix-jvm-sdk/tree/" + branch
 }
 
-lazy val devTools = project
-  .in(file("devtools"))
-  .enablePlugins(BuildInfoPlugin, PublishSonatype)
-  .dependsOn(coreSdk)
-  .settings(common)
-  .settings(
-    name := "kalix-devtools",
-    Compile / javacOptions ++= Seq("--release", "11"),
-    Compile / scalacOptions ++= Seq("-release", "11"),
-    scalaVersion := Dependencies.ScalaVersionForDevTools,
-    buildInfoKeys := Seq[BuildInfoKey](
-      name,
-      version,
-      "proxyImage" -> "gcr.io/kalix-public/kalix-proxy",
-      "proxyVersion" -> Kalix.ProxyVersion,
-      "scalaVersion" -> scalaVersion.value),
-    buildInfoPackage := "kalix.devtools",
-    // Generate javadocs by just including non generated Java sources
-    Compile / doc / sources := {
-      val javaSourceDir = (Compile / javaSource).value.getAbsolutePath
-      (Compile / doc / sources).value.filter(_.getAbsolutePath.startsWith(javaSourceDir))
-    },
-    // javadoc (I think java 9 onwards) refuses to compile javadocs if it can't compile the entire source path.
-    // but since we have java files depending on Scala files, we need to include ourselves on the classpath.
-    Compile / doc / dependencyClasspath := (Compile / fullClasspath).value,
-    Compile / doc / javacOptions ++= Seq(
-      "-Xdoclint:none",
-      "-overview",
-      ((Compile / javaSource).value / "overview.html").getAbsolutePath,
-      "-notimestamp",
-      "-doctitle",
-      "Kalix Dev Tools",
-      "-noqualifier",
-      "java.lang"))
-  .settings(Dependencies.devTools)
+lazy val devTools = devToolsCommon(
+  project
+    .in(file("devtools"))
+    .settings(name := "kalix-devtools", scalaVersion := Dependencies.ScalaVersion))
+
+/*
+  This variant devTools compiles with Scala 2.12, but uses the same source files as the 2.13 version (above).
+  This is needed because the devTools artifact is also used by the sbt and mvn plugins and therefore needs a 2.12 version.
+  Note that crossbuilding is not an option, because when the built selects 2.13, it will try to build sbt-kalix with 2.13
+ */
+lazy val devToolsInternal =
+  devToolsCommon(
+    project
+      .in(file("devtools"))
+      .settings(
+        name := "kalix-devtools-internal",
+        scalaVersion := Dependencies.ScalaVersionForTooling,
+        // to avoid overwriting the 2.13 version
+        target := baseDirectory.value / "target-2.12"))
```

Comment:
Maybe some more insight on this.

We want `devTools` to be compiled to both 2.12 and 2.13 so we can use it in integration tests and not only as a dependency of mvn and sbt plugins. 

If we make it a normal cross-build and compile with `+compile`, sbt will first compile everything for 2.12. Meaning dev-tools and the sbt-kalix plugin. 

Then it compiles for 2.13. In that second run, it will build devtools_2.13, but when it reaches the sbt-kalix plugin, it switches back to 2.12 because the plugin requires 2.12. At that moment, we get a Scala version conflict. 

A solution to this problem is to create two modules and let the sbt plugin only depend on `kalix-devtools-internal`. 

My first attempt was to redefine the `sourceDirectory` in the internal module to point to the `devTools` src. Surprisingly enough that didn't work in CI. On Linux, somehow we end up with an almost empty jar (contained BuildInfo only). 

My second attempt, which proved to work, was to at a specific target folder. Hence the `target-2.12` hack here.

Reply:
Note that `devTools` and `devToolsInternal` modules are sharing the same directory. `in(file(devtools))`, but using different `scalaVersion` and the latter has a different `target` folder.

Reply:
Would it make sense to name them `devTools213` and `devTools212`?

Reply:
I did it at some point, but came to the conclusion that it was easier to follow if we said: `devTools` (for general usage) and `devToolsInternal` to be used internally by our own plugins. 

Note that these are only the name of sbt modules. 

Users will eventually see `kalix-devtools_2.13` and  `kalix-devtools-internal_2.12`. And in that case, it's better to have 'internal' in the main. It kind of signals that this is not for general usage.